tests/dataset/test_aoi.py

```diff
@@ -81,6 +81,65 @@ def test_bounds_iou(self) -> None:
         expected_iou = 0.013904645827033404
         assert iou == expected_iou
 
+    def test_corrupt_raster(self) -> None:
+        raster_file = "tests/data/massachusetts_buildings_kaggle/22978945_15_uint8_clipped_clipped.tif"
+        with pytest.raises(BaseException):
+            raster = rasterio.open(raster_file)
+
+    def test_image_only(self) -> None:
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/sampling/sampling_segmentation_binary-multiband_ci.csv")
+        for row in data:
+            aoi = AOI(raster=row['tif'], label=None)
+
+    def test_missing_label(self) -> None:
+        extract_archive(src="tests/data/spacenet.zip")
+        data = read_csv("tests/sampling/missing_label.csv")
+        for row in data:
+            with pytest.raises(rasterio.errors.RasterioIOError):
+                aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'])
```

Comment:
This one is also good (except for comment above), but there should be a more explicit errors raised in the aoi creation.
Right after the missing label file is "read", the function _check_gdf_load() returns an empty GeoDataFrame. Therefore, after [this line](https://github.com/NRCan/geo-deep-learning/blob/develop/dataset/aoi.py#L223), there should be something like:
`if self.label_gdf.empty:
     raise fiona.errors.DriverError(f"Fiona could not load label file: {label}")`